### PR TITLE
MNT Add support for symphony 4 and switch to shared travis config

### PR DESCRIPTION
**Pull request by maxime-rainville**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -27,10 +27,10 @@
         "behat/mink": "^1.7",
         "behat/mink-extension": "^2.1",
         "silverstripe/mink-facebook-web-driver": "^1",
-        "symfony/dom-crawler": "^3",
+        "symfony/dom-crawler": "^3|^4",
```
Comment (emteknetnz) [inline composer.json]: Even though this works, standard syntax is "^3 || ^4"

https://getcomposer.org/doc/articles/versions.md#version-range

https://github.com/silverstripe/silverstripe-asset-admin/blob/1/composer.json#L9

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Looks good, just need a  minor change on the version constraint syntax

File: composer.json (hunk 1 above)
Comment (maxime-rainville) [reply composer.json]:
```suggestion
        "symfony/dom-crawler": "^3||^4",
```

File: composer.json [hunk 2]
```diff
@@ -27,10 +27,10 @@
...
         "silverstripe/testsession": "^2.2",
         "silverstripe/framework": "^4",
-        "symfony/finder": "^3.2"
+        "symfony/finder": "^3.2|^4"
```
Comment (maxime-rainville) [inline composer.json]:
```suggestion
        "symfony/finder": "^3.2||^4"
```

**Comment by maxime-rainville** [comment]
One day, I will get those annoying OR composer constraint correct on the first try.

File: composer.json [hunk 3]
```diff
@@ -27,10 +27,10 @@
         "behat/mink": "^1.7",
         "behat/mink-extension": "^2.1",
         "silverstripe/mink-facebook-web-driver": "^1",
-        "symfony/dom-crawler": "^3",
+        "symfony/dom-crawler": "^3||^4",
```
Comment (emteknetnz) [inline composer.json]:
```suggestion
        "symfony/dom-crawler": "^3 || ^4",
```

File: composer.json [hunk 4]
```diff
@@ -27,10 +27,10 @@
...
         "silverstripe/testsession": "^2.2",
         "silverstripe/framework": "^4",
-        "symfony/finder": "^3.2"
+        "symfony/finder": "^3.2||^4"
```
Comment (emteknetnz) [inline composer.json]:
```suggestion
        "symfony/finder": "^3.2 || ^4"
```

**Review by emteknetnz — CHANGES_REQUESTED** [review]
:-)

**Pull request merged**